app/models/katello/host/subscription_facet.rb

```diff
@@ -35,6 +39,35 @@ def import_database_attributes(consumer_params)
         end
       end
 
+      def update_hypervisor(consumer_params = candlepin_consumer.consumer_attributes)
+        if consumer_params.try(:[], 'type').try(:[], 'label') == 'hypervisor'
+          self.hypervisor = true
+        elsif !consumer_params.try(:[], 'guestIds').empty?
+          self.hypervisor = true
+        elsif !candlepin_consumer.virtual_guests.empty?
+          # Check by calling out to Candlepin last for efficiency
+          self.hypervisor = true
+        end
+      end
+
+      def update_guests(consumer_params)
+        if self.hypervisor
+          if !consumer_params.try(:[], 'guestIds').empty?
+            guest_ids = consumer_params['guestIds'].map{ |i| i['guestId'].downcase}
```

Comment:
So this is a case where the Katello API is not mimicking the RHSM API in terms of the data format it expects? Or can you point me to an example where `virt-who` provided the `guestIds` field as an array of strings?